File: backend/api/models/message.py
from django.db import models
from .conversation import Conversation

class Message(models.Model):
    question = models.TextField()
    answer = models.TextField(null=True, blank=True)
    liked_at = models.DateTimeField(null=True, blank=True)
    disliked_at = models.DateTimeField(null=True, blank=True)
    answered_at = models.DateTimeField(null=True, blank=True)
    failed_at = models.DateTimeField(null=True, blank=True)
    sources = models.JSONField(null=True, blank=True)
    conversation = models.ForeignKey(
        Conversation, on_delete=models.CASCADE, related_name="messages"
    )
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        db_table = "messages"
        ordering = ["id"]

File: frontend/src/interfaces/message.ts
export interface Message {
  id: number;
  question: string;
  answer: string;
  liked_at: string;
  disliked_at: string;
  answered_at: string;
  failed_at: string;
  sources: JSON;
  conversation_id: number;
  created_at: string;
  updated_at: string;
}
